src/overlays/actors/ovl_Bg_Iknv_Obj/z_bg_iknv_obj.c

```diff
@@ -21,32 +31,190 @@ const ActorInit Bg_Iknv_Obj_InitVars = {
     (ActorFunc)BgIknvObj_Update,
     (ActorFunc)BgIknvObj_Draw,
 };
-*/
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_Init.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_Destroy.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD7CEC.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateType0.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD7E0C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD7ED8.asm")
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD7F4C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD7FDC.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/func_80BD8040.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateType2.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateType1.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_UpdateDefault.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_Update.asm")
+static ColliderCylinderInit sCylinderInit = {
+    { 
+        COLTYPE_NONE, 
+        AT_NONE,
+        AC_ON | AC_TYPE_ENEMY, 
+        OC1_ON | OC1_TYPE_ALL, 
+        OC2_TYPE_1, 
+        COLSHAPE_CYLINDER, 
+    },
+    { 
+        ELEMTYPE_UNK0, 
+        { 0x00000000, 0x00, 0x00 }, 
+        { 0xF7CFFFFF, 0x00, 0x00 }, 
+        TOUCH_NONE | TOUCH_SFX_NORMAL, 
+        BUMP_ON, 
+        OCELEM_ON, 
+    },
+    { 40, 40, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Iknv_Obj_0x80BD7AB0/BgIknvObj_Draw.asm")
+void BgIknvObj_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIknvObj* this = THIS;
+    CollisionHeader* colHeader = NULL;
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    this->actionFunc = BgIknvObj_DoNothing;
+    switch (IKNV_OBJ_TYPE(this)) {
+        case IKNV_OBJ_WATERWHEEL: {
+            this->displayListPtr = D_06013058;
+            this->actionFunc = BgIknvObj_UpdateWaterwheel;
+            this->dyna.actor.flags |= 0x100000;
+            this->dyna.actor.flags |= 0x10;
+            return;
+        }
```

Comment:
```suggestion
        case IKNV_OBJ_WATERWHEEL:
            this->displayListPtr = D_06013058;
            this->actionFunc = BgIknvObj_UpdateWaterwheel;
            this->dyna.actor.flags |= 0x100000;
            this->dyna.actor.flags |= 0x10;
            return;
```
You don't need curly braces between each case. Only one for the entire switch